runtime/coverage.go

```diff
@@ -173,6 +177,22 @@ func (r *CoverageReport) InspectProgram(location Location, program *ast.Program)
 					line := element.StartPosition().Line
 					lineHits[line] = 0
 				}
+
+				functionBlock, isFunctionBlock := element.(*ast.FunctionBlock)
+				// Track also pre/post conditions defined inside functions.
+				if isFunctionBlock {
+					conditions := make([]*ast.Condition, 0)
+					if functionBlock.PreConditions != nil {
+						conditions = append(conditions, *functionBlock.PreConditions...)
+					}
+					if functionBlock.PostConditions != nil {
+						conditions = append(conditions, *functionBlock.PostConditions...)
+					}
+					for _, condition := range conditions {
+						line := condition.Test.StartPosition().Line
+						lineHits[line] = 0
+					}
```

Comment:
Maybe avoid the allocation of a temporary slice here, and avoid the code duplication of getting the start position line and recording it in the line hit map. 

You can do this e.g. by defining a helper function, after `lineHits` is introduced (see above, line 154):

```go
  	recordLine := func(hasPosition ast.HasPosition) {
		line := hasPosition.StartPosition().Line
		lineHits[line] = 0
	} 
```

Replace the recording for the statement above (line 173):

```go
					recordLine(element)
```

Then use it for the conditions:

```suggestion
					if functionBlock.PreConditions != nil {
						for _, condition := range *functionBlock.PreConditions {
							recordLine(condition.Test)
						}
					}

					if functionBlock.PostConditions != nil {
						for _, condition := range *functionBlock.PostConditions {
							recordLine(condition.Test)
						}
					}
```

Reply:
Great suggestion :clap:
Updated in https://github.com/onflow/cadence/pull/2371/commits/bb43f13aba26f9916436ea96ae31c9dc30ab7327